components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -71,6 +77,38 @@ type I2CConfig struct {
 // Validate ensures all parts of the config are valid.
 func (cfg *Config) Validate(path string) ([]string, error) {
 	var deps []string
+
+	dep, err := cfg.validateCorrectionSource(path)
+	if err != nil {
+		return nil, err
+	}
+	if dep != nil {
+		deps = append(deps, dep...)
+	}
+
+	dep, err = cfg.validateConnectionType(path)
+	if err != nil {
+		return nil, err
+	}
+	if dep != nil {
+		deps = append(deps, dep...)
+	}
+
+	if cfg.CorrectionSource == ntripStr {
+		dep, err = cfg.validateNtripInputProtocol(path)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if dep != nil {
```

Comment:
I don't think deps is ever nil since it's being created in line 79, do can you remove this if statement?

Reply:
I think it can be nil if the input protocol isn't i2c since that's the only one that has dependencies

Reply:
Yeah that makes sense, but when it calls validateCorrectionSouce, validateConnectionType, and validateNtripProtocol, if any of those config values are i2c then the board will be returned as a dep but if its serial/ntrip then they would just return nil, so if any dependency is returned it will be added to the list. But definitely agree this can still be cleaned up, might be better to just move all the dependency logic directly to the ValidateI2C function. I made a ticket for improving the validation functions and Susmita is currently working on it so I will let her know!